lib/full-site-editing/template-loader.php

```diff
@@ -291,3 +300,34 @@ function gutenberg_get_edit_template_link( $link, $post_id ) {
 	return admin_url( sprintf( $edit_link, urlencode( $template->id ), $template->type ) );
 }
 add_filter( 'get_edit_post_link', 'gutenberg_get_edit_template_link', 10, 2 );
+
+/**
+ * Checks the HTML to be rendered for any emojis.
+ * If there are none, then the emoji script is removed from the `<head>`.
+ *
+ * @param string $html The HTML to be rendered.
+ *
+ * @return string The HTML.
+ */
+function _gutenberg_maybe_remove_emoji_detection_script( $html ) {
+	$multi3  = '(?:[\xE0-\xEF]..)';
+	$multi4  = '(?:[\xF0-\xF4]...)';
+	$anychar = "(?:$multi4|$multi3)";
+
+	// Detect all 3-byte and 4-byte characters.
+	// Not all 3/4-byte characters are emojis, but this casts a wide net
+	// to check if it's safe to remove the emojis script.
+	// This regex should be improved in the future to be more precice,
+	// but even this wide net should be enough to remove the script
+	// in a lot of sites that don't need it.
+	preg_match( "/$anychar/", $html, $match );
+
+	// If there are no 4-byte characters, it's safe to remove the emoji script.
+	if ( empty( $matches ) ) {
```

Comment:
The variable name is wrong here :). Also I guess we can make the check with the returned value of `preg_match`.

Reply:
LOL dumb typo... Thanks for catching it!